### PR TITLE
support common input for operational input on Maintenance Management API 

### PR DESCRIPTION
**Pull request by xyuanlu**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

#2034

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

This change adds a common input field for operational check input on Maintenance Management API. 

### Tests

- [X] The following tests are written for this issue:

testTakeInstanceOperationCheckFailureCommonInput

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java [hunk 1]
```diff
@@ -361,13 +364,20 @@ private MaintenanceManagementInstanceInfo takeFreeSingleInstanceHelper(String cl
...
+
       // perform operation check
       for (OperationInterface operationClass : operationAbstractClassList) {
         String operationClassName = operationClass.getClass().getName();
         Map<String, String> singleOperationConfig =
             (operationConfig == null || !operationConfig.containsKey(operationClassName))
                 ? Collections.emptyMap()
                 : getMapFromJsonPayload(operationConfig.get(operationClassName));
+        commonOperationConfig
+            .forEach((key, value) -> singleOperationConfig.merge(key, value, (v1, v2) -> v1));
```
Comment (qqu0127) [inline helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java:380]: It seems we always keep the value in `singleOperationConfig`? 
If so, could we use `commonOperationConfig.forEach(singleOperationConfig::putIfAbsent)`? Let's also add a comment here to clarify the combining logic.

Comment (xyuanlu) [reply helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java]: TFTR. Good point. Updated.

**Review by desaikomal — COMMENTED** [review]
LGTM - thanks for working on this.

**Comment by xyuanlu** [comment]
This PR is ready to be merged, approved by @desaikomal //Thanks!
Commit message:
support common input for operational input on Maintenance Management API

**Pull request merged**